### PR TITLE
build: cut precommit hook, introduce proto-change test

### PR DESCRIPTION
**Pull request by dmccartney**

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env sh
-. "$(dirname -- "$0")/_/husky.sh"
-
-npx commitlint --edit
```
Comment (neekolas) [inline .husky/commit-msg:4]: We still need this to enforce the commit message conventions, which are used by semantic-release.

**Review by neekolas — CHANGES_REQUESTED** [review]
I'm good with removing the build hook, but we still need the commit message hook. Otherwise commits won't generate releases on NPM.

File: .husky/commit-msg (hunk 1 above)
Comment (neekolas) [reply .husky/commit-msg:4]: Now that this is going to be generating code for more platforms, we are probably going to have to rethink our conventions for semantic-release (should a breaking change to the Dart build really trigger a new release on the JS package? Probably not). But I think that for the time-being we should leave things as-is and update when we have a reasonable alternative.

**Review by neekolas — APPROVED** [review]

**Comment by dmccartney** [comment]
This isn't quite working: the `protoc` for `generate.go` inside the `proto-change-test` workflow is having versioning and path issues (so it's giving us false-negatives -- it's failing when a proto file changes but not because of an unexpected diff). 

Instead of fighting through the go env quirks of this, I'm inclined to close this and revisit when time permits us to refactor codegen more generally. (But if someone else is inspired, feel free to pick this up and run with it.)

**Pull request closed without merging**